web/src/main/java/co/mobileaction/example/web/service/UserService.java

```diff
@@ -0,0 +1,39 @@
+package co.mobileaction.example.web.service;
+
+import co.mobileaction.example.web.exception.AlreadyExistException;
+import co.mobileaction.example.web.model.User;
+import co.mobileaction.example.web.repository.IUserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author elif
+ * @date 05.07.2023
+ * @time 14.37
+ */
+@Service
+@RequiredArgsConstructor
+public class UserService implements IUserService
+{
+    private final IUserRepository userRepository;
+
+    @Override
+    public void saveUser(User user) throws AlreadyExistException
+    {
+        // check if user already exists
+        boolean userAlreadyExists = userExist(user.getId());
```

Comment:
No need assign to variable. You can user directly in if block.
```suggestion
        if (userExist(user.getId()))
```

Reply:
I think assigning to a new variable looks more readable